external/storm-kafka-client/src/main/java/org/apache/storm/kafka/spout/trident/KafkaTridentSpoutEmitter.java

```diff
@@ -60,6 +65,14 @@
     private static final long serialVersionUID = -7343927794834130435L;
     private static final Logger LOG = LoggerFactory.getLogger(KafkaTridentSpoutEmitter.class);
 
+    // Metrics
+    public static final String UNDERSCORE = "_";
+    public static final String INTERNAL_KAFKA_RECORDS_LAG_MAX_METRIC = "records-lag-max";
+    public static final String KAFKA_CLIENT_MAX_LAG_METRIC_NAME = "kafkaClientMaxLag";
+    protected transient Gauge<Double> kafkaClientMaxLag;
+    public static final String EVENT_EMIT_METRIC_NAME = "eventEmitRate";
+    protected transient Meter eventEmitRate;
```

Comment:
Is the builtin emitted metric not sufficient for this purpose? https://github.com/apache/storm/blob/master/docs/Metrics.md#__emit-count

Reply:
This builtin might be sufficient.  Our metric names seem to be reporting values like `__emit-count-s1` or `__emit-count-s2` (there are almost 200 of these on our Yamas metric search, I'm guessing one per stream).  Perhaps something is mis-configured in our topologies preventing it from extracting useful stream names.

Reply:
With the recent change on metrics in storm (converting to v2 metrics),  stream name will be appended to the metric name. 

```
The tuple counting metric names contain "${stream_name}" or "${upstream_component}:${stream_name}". The former is used for all spout metrics and for outgoing bolt metrics (__emit-count and __transfer-count). The latter is used for bolt metrics that deal with incoming tuples.
```


[DimensionalReporter](https://github.com/apache/storm/blob/master/storm-client/src/jvm/org/apache/storm/metrics2/DimensionalReporter.java) can be used to separate dimensions (stream_name, componentId, etc) from metrics.